terraform/cloud-platform-components/README.md

```diff
@@ -82,6 +83,114 @@ Metrics-server allows us to perform resource queries against the cluster. Comman
+
+We utilise [opa-helm-chart](https://github.com/helm/charts/tree/master/stable/opa) to deploy opa onto the Cloud Platform. This helm chart installs OPA as a Kubernetes admission controller. In our case we are using validating admission controller. The helm Chart will automatically generate a CA and server certificate for the OPA. 
+
+In Kubernetes, Admission Controllers enforce semantic validation of objects during create, update, and delete operations. With OPA we can enforce custom policies on Kubernetes objects without recompiling or reconfiguring the Kubernetes API server. Please see the following documentation by opa on Kubernetes Admission Control 
+https://www.openpolicyagent.org/docs/kubernetes-admission-control.html
+
+To restrict the kinds of operations and resources that are subject to OPA policy checks, see the below example. Configured in templates/opa/values.yaml.tpl under admissionControllerRules:
+
+```yaml
+    admissionControllerRules:
+      - operations: ["CREATE", "UPDATE"]
+        apiGroups: ["extensions", ""]
+        apiVersions: ["v1beta1”, “v1”]
+        resources: ["ingresses", "namespaces"]
+```
+The admissionControllerRules the operations and resources that the webhook will validate. Intercept API requests when a ingress or a namespace is CREATED or UPDATED, so apiGroups and apiVersions are filled out accordingly (extensions/v1beta1 for ingresses, v1 for namespaces). We can use wildcards (*) for these fields as well.
+
+ ### kube-mgmt
+
+kube-mgmt manages instances of the Open Policy Agent on top of kubernetes. Use kube-mgmt to:
+
+    - Load policies into OPA via kubernetes.
+    - Replicate kubernetes resources including CustomResourceDefinitions (CRDs) into OPA.
+
+ ### OPA-Policies
+
+Policies are written in a high-level declarative language and can be loaded into OPA via the filesystem or well-defined APIs.
+
+kube-mgmt automatically discovers policies stored in ConfigMaps in kubernetes and loads them into OPA. kube-mgmt assumes a ConfigMap contains policies if the ConfigMap is:
+
+    - Created in a namespace listed in the --policies option. Configured in templates/opa/values.yaml.tpl under mgmt/configmapPolicies/namespaces:
+	  - Labelled with openpolicyagent.org/policy=rego.
+
+When a policy has been successfully loaded into OPA, the openpolicyagent.org/policy-status annotation is set to {"status": "ok"}.
+If loading fails for some reason (e.g., because of a parse error), the openpolicyagent.org/policy-status annotation is set to {"status": "error", "error": ...} where the error field contains details about the failure.
+
+IMP NOTE:  Apply a ConfigMap that contains the main OPA policy and default response. This policy is used as an entry-point for policy evaluations and returns allowed:true if policies are not matched to inbound data.
```

Comment:
Can we please mention the file where this entrypoint is defined?

Reply:
mentioned the file information

Reply:
pointed to opa-default-system-main.yaml which contains the main OPA policy and used as an entry-point for policy evaluations.

Reply:
Sweet! Love the link ^_^